### ISSUE TITLE
Plugin-kafka-publish randomly dropping contexts

### ISSUE THREAD

**Issue by fpguy** [ovh/cds#2861]
To summarize, what happen is the following:
1./  Sending context #CONTEXT_NUMBER with plugin-kafka on CDS
2./ Getting the following warning on my instance side (plugin-kafka-publish)
```bash
Listening Kafka $ENDPOINT on topic $TOPIC...
Unknown CDS context : $CONTEXT_NUMBER&filename=file.tar.gz
# ( .... 100 lines)
Unknown CDS context : $CONTEXT_NUMBER&filename=file.tar.gz 
```

In such a case, please be aware that:
 - context #CONTEXT_NUMBER has never been received on my instance side
 - context #CONTEXT_NUMBER artifact (file.tar.gz) has been dropped away (as mentioned above)
 - kafka topic is not in error state (all content consumed)
 - sending a new context is not enough, a proper ack (valid expected context number) must be received on CDS to unlock this blocking state and then be able to get a complete context

That's why I have created a "emergency_ack.sh" script which takes the context ID to create a context_file.json and answers it through -ack topic.
This solution is everything but beautiful however as the issue was very rare it didn't bother me.
 
However for around 1 week I had to use it at least twice which is WAY too much...!

Additional note about randomness of this phenomenon: it seems to happen in particular when kafka is being used again after 2/3 days of pause (typically on Mondays morning!)

**Comment by fpguy** [comment]
As discussed with @yesnault the issue seems to be caused by the automated deletion of the kafka group after a few days of inactivity (typically a weekend).
The group seems to be created again only when a listener connects to the topic with the group again.

To fix the issue in our case we decided to wait on kafka API until the group shows up again, meaning that a listener is ready to receive our context from the beginning.

=> Embedding this check in the plugin-kafka-publish on CDS, might be a good enhancement. With either a blocking option (with a timeout) or not (exit fail).

The quick-fix has been merged today on our side, so let's see how the queue behaves on next Mondays.

Thanks for your inputs!

**Comment by yesnault** [comment]
seen both, no need anymore - won't fix.
